### PR TITLE
feat: implement dagger

### PR DESCRIPTION
**Pull request by lrojas870**
- dagger library implementation
- creation of the ApplicationComponent class
- module creation

### CODE REVIEW

File: app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/NetworkModule.kt [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        return builder.client(okHttpClient).build()
+    }
+
+    @Provides
+    fun provideServicesInterface(@Named("provideRetrofit") retrofit: Retrofit) : ServicesApiInterface {
+
+        var servicesApiInterface: ServicesApiInterface?=null
+        servicesApiInterface = retrofit.create(ServicesApiInterface::class.java)
+        return servicesApiInterface as ServicesApiInterface
```
Comment (danponce) [inline app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/NetworkModule.kt]: Why not return retrofit.create(ServicesApiInterface::class.java) ?

File: app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/RepositoryModule.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.androidavanzado.covid19stadistics.model.Repository
+import com.androidavanzado.covid19stadistics.storage.remote.RemoteDataSource
+import com.androidavanzado.covid19stadistics.storage.remote.ServicesApiInterface
+import dagger.Module
+import dagger.Provides
+import javax.inject.Singleton
+
+
+/*object Injector {
```
Comment (danponce) [inline app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/RepositoryModule.kt]: if this is not going to be used in the future better remove it 👍

File: app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt [hunk 3]
```diff
@@ -2,50 +2,49 @@ package com.androidavanzado.covid19stadistics.ui
...
-import com.bumptech.glide.load.engine.Resource
-import java.lang.reflect.Array.set
 import java.text.NumberFormat
+import javax.inject.Inject
 
 class HomeActivity : AppCompatActivity() {
 
     private lateinit var binding : ActivityHomeBinding
-    private lateinit var viewModel:StadisticsViewModel
+    private lateinit var myApplication: MyApplication
```
Comment (danponce) [inline app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt]: this var is not necessary. You can simply call inside onCreate like this: MyApplication().appComponent.inject(this)

**Comment by danponce** [comment]
too few commits for so many changes. Next time try to split those changes in more commits 👍

File: app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/NetworkModule.kt (hunk 1 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/NetworkModule.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/RepositoryModule.kt (hunk 2 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/dependency/module/RepositoryModule.kt]: done

File: app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt (hunk 3 above)
Comment (lrojas870) [reply app/src/main/java/com/androidavanzado/covid19stadistics/ui/HomeActivity.kt]: done

**Review by danponce — APPROVED** [review]

**Pull request merged**